File: Python/PerfectNumber/PerfectNumber.py
import math

class Solution(object):
    # (62ms)
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        return sum([i + num / i for i in xrange(2, int(math.sqrt(num)) + 1) if num % i == 0]) + 1 == num if num > 1 else False

File: Java/PerfectNumber/PerfectNumber.java
/**
 * (15ms)
 * 这可能是一个假的medium题。
 * @author Phoenix-Z
 *
 */
public class PerfectNumber {

	public static boolean checkPerfectNumber(int num) {
		if(num == 1) {
			return false;
		}
        int end = (int) Math.sqrt(num);
        int sum = 1;
        for(int i = 2; i <= end; i++) {
        	if(num % i == 0) {
        		sum += i + num / i;
        	}
        }
        return sum == num;
    }
	
	public static void main(String[] args) {
		System.out.println(checkPerfectNumber(28));
	}
	
}
